ReactWindows/ReactNative.Shared/UIManager/BorderedCanvas.cs

```diff
@@ -47,6 +52,298 @@ public Border Border
+        }
+
+        /// <summary>
+        /// Gets or sets background brush of Canvas itself or of Border child as
+        /// appropriate. This is the only supported method to access background
+        /// of BorderedCanvas. DO NOT use Canvas.Background or Border.Background directly.
+        /// </summary>
+        public new Brush Background
+        {
+            get
+            {
+                if (Border == null)
+                {
+                    return base.Background;
+                }
+                else
+                {
+                    var border = GetOrCreateBorder();
+                    return border.Background;
+                }
+            }
+            set
+            {
+                if (Border == null)
+                {
+                    base.Background = value;
+                }
+                else
+                {
+                    var border = GetOrCreateBorder();
+                    border.Background = value;
+                    base.Background = base.Background ?? _defaultBackgroundBrush;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the border radius of the view.
+        /// </summary>
+        public CornerRadius BorderRadius
+        {
+            get
+            {
+                if (HasBorder())
+                {
+                    var border = GetOrCreateBorder();
+                    return border.CornerRadius;
+                }
+                else
+                {
+                    return new CornerRadius();
+                }
+            }
+            set
+            {
+                var border = GetOrCreateBorder();
+                border.CornerRadius = value;
+                TransferBackgroundBrush();
+            }
+        }
+
+        /// <summary>
+        /// Sets the background color of the view.
+        /// </summary>
+        public Color BackgroundColor
+        {
+            set => Background = new SolidColorBrush(value);
+        }
+
+        /// <summary>
+        /// Set the border color of the view.
+        /// </summary>
+        public Color? BorderColor
+        {
+            set
+            {
+                if (Border == null)
+                {
+                    GetOrCreateBorderProps().Color = value;
+                }
+                else
+                {
+                    var border = GetOrCreateBorder();
+                    border.BorderBrush = value.HasValue
+                        ? new SolidColorBrush(value.Value)
+                        : DefaultBorderBrush;
+                    TransferBackgroundBrush();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the border thickness of the view.
+        /// </summary>
+        /// <param name="index">The property index.</param>
+        /// <param name="width">The border width in pixels.</param>
+        public void SetBorderWidth(int index, double width)
```

Comment:
>SetBorderWidth [](start = 20, length = 14)

Any reason this shouldn't be a property?

Reply:
It has two parameters.

---
In reply to: [178548821](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r178548821) [](ancestors = 178548821)

Reply:
Can we just expose the BorderThickness struct?